metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -180,28 +189,48 @@ func (m *metricHints) getMetricSets(hints common.MapStr, module string) []string
 	return msets
 }
 
-func (m *metricHints) getHostsWithPort(hints common.MapStr, port int) ([]string, bool) {
+func (m *metricHints) getHostsWithPort(hints common.MapStr, port int, noPort bool) ([]string, bool) {
 	var result []string
 	thosts := builder.GetHintAsList(hints, m.Key, hosts)
 
 	// Only pick hosts that have ${data.port} or the port on current event. This will make
 	// sure that incorrect meta mapping doesn't happen
 	for _, h := range thosts {
-		if strings.Contains(h, "data.port") || strings.Contains(h, fmt.Sprintf(":%d", port)) ||
+		if (strings.Contains(h, "data.port") && port != 0 && !noPort) || m.checkHostPort(h, port) ||
```

Comment:
Instead of this `checkHostPort`, whould `strings.HasSuffix` do the same?
```suggestion
		if (strings.Contains(h, "data.port") && port != 0 && !noPort) || strings.HasSuffix(h, fmt.Sprintf(":%d", port)) ||
```

Reply:
this wont work as a lot of hints have api paths as well. hence why this check is done.